src/nupic/types/Serializable.hpp

```diff
@@ -273,11 +287,17 @@ class Serializable {
   }
   
   virtual inline void loadFromFile_ar(std::string filePath, SerializableFormat fmt=SerializableFormat::BINARY) {
-    std::ifstream in(filePath, std::ios_base::in | std::ios_base::binary);
-    in.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+		std::ios_base::openmode mode = std::ios_base::in;
+		if (fmt <= SerializableFormat::PORTABLE) mode |= std::ios_base::binary;
+	  std::ifstream in(filePath, mode);
+    // NOTE: do NOT set stream exceptions:
+    //       in.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+    //       The JSON parser will not be able to find the end of the parse.
```

Comment:
OT: can you file a bug at cereal?

Reply:
> The JSON parser will not be able to find the end of the parse.
OT: can you file a bug at cereal?

It is not really a bug.  The JSON parser knows that it is at the end when it reads a character and gets a null.  If we turn on stream exceptions it throws an exception at that point and everything stops. I know this is not good design but this is not Cereal's code either.